app/core/event_handler.py

```diff
@@ -13,38 +13,40 @@
 async def handle_event(premis_event: PremisEvent) -> None:
     log.debug(
         "Start handling of PREMIS event.",
-        fragment_id=premis_event.fragment_id,
+        mediahaven_id=premis_event.mediahaven_id,
     )
 
     mediahaven_service = MediahavenService(config.app_cfg)
-    fragment_id = premis_event.fragment_id
+    mediahaven_id = premis_event.mediahaven_id
 
     # Get metadata for the newly archived item
     try:
-        fragment = mediahaven_service.get_fragment(fragment_id)
-        original_pid = fragment["Dynamic"]["s3_object_key"].split(".")[0]
+        fragment = mediahaven_service.get_fragment(mediahaven_id)
     except MediaObjectNotFoundException as e:
         log.critical(
-            "Got a PREMIS event, but the fragment id is not in MediaHaven",
-            fragment_id=premis_event.fragment_id,
+            "Got a PREMIS event, but the mediahaven id is not in MediaHaven",
+            mediahaven_id=premis_event.mediahaven_id,
             exception=str(e),
         )
         return
+
+    try:
+        original_pid = fragment["Dynamic"]["s3_object_key"].split(".")[0]
+        fragment_id = fragment["Internal"]["FragmentId"]
```

Comment:
The `FragmentId` should not or can not ever be missing on the item. So I would suggest:

```suggestion

    fragment_id = fragment["Internal"]["FragmentId"]
    try:
        original_pid = fragment["Dynamic"]["s3_object_key"].split(".")[0]
```

But, better leave it like this as I don't want to be proven wrong. ;-)